### PR TITLE
Update to bitvec 0.22

### PR DESCRIPTION
**Pull request by nuttycom**
- MSRV is now 1.51.0.
- A new `bits` feature flag exposes an `ff::PrimeFieldBits` implementation on `jubjub::Scalar`.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -59,3 +60,8 @@ harness = false
 [[bench]]
 name = "point_bench"
 harness = false
+
+[patch.crates-io]
+bls12_381 = { git = "https://github.com/nuttycom/bls12_381.git", branch = "bitvec-0.22", default-features = false }
```
Comment (str4d) [inline Cargo.toml:65]: Once https://github.com/zkcrypto/bls12_381/pull/62 merges:
```suggestion
bls12_381 = { git = "https://github.com/zkcrypto/bls12_381.git", branch = "main", default-features = false }
```

File: src/lib.rs [hunk 2]
```diff
@@ -1231,6 +1231,10 @@ impl CofactorGroup for ExtendedPoint {
     fn into_subgroup(self) -> CtOption<Self::Subgroup> {
         CtOption::new(SubgroupPoint(self), self.is_torsion_free())
     }
+
+    fn is_torsion_free(&self) -> Choice {
+        self.is_torsion_free()
+    }
```
Comment (ebfull) [inline src/lib.rs:1237]: Are we sure this isn't recursing forever and nothing is calling it in our test suite? Might be better to do `ExtendedPoint::is_torsion_free(self)`.

**Review by ebfull — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 2 above)
Comment (str4d) [reply src/lib.rs:1237]: It's a call to a concrete method inside a trait method, which rustc does know how to handle. The stack overflow issue we saw in the Zcash crates was due to two traits having the same method name.

Comment (str4d) [reply src/lib.rs:1237]: From https://doc.rust-lang.org/book/ch19-03-advanced-traits.html#fully-qualified-syntax-for-disambiguation-calling-methods-with-the-same-name:

> When we call `fly` on an instance of `Human`, the compiler defaults to calling the method that is directly implemented on the type
> [...]
> Specifying the trait name before the method name clarifies to Rust which implementation of `fly` we want to call. We could also write `Human::fly(&person)`, which is equivalent to the `person.fly()` that we used in Listing 19-18, but this is a bit longer to write if we don’t need to disambiguate.

Comment (ebfull) [reply src/lib.rs:1237]: Okay, cool.

**Review by ebfull — APPROVED** [review]

File: Cargo.toml [hunk 3]
```diff
@@ -47,6 +47,7 @@ default-features = false
 [features]
 default = ["alloc"]
```
Comment (str4d) [inline Cargo.toml:48]:
```suggestion
default = ["alloc", "bits"]
```

**Pull request merged**